File: Python/BinaryTreeRightSideView.py
from collections import deque

class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def binaryTreeRightSideView(root):
        outputList = []
        queue = deque()
        levelSet = set()

        level = 0

        if root:
            queue.append((root,level))

        while len(queue)>0:
            level += 1
            for i in range(len(queue)):
                curr,levelToCheck = queue.popleft()
                if levelToCheck not in levelSet:
                    outputList.append(curr.val)
                    levelSet.add(levelToCheck)

                if curr.right:
                    queue.append((curr.right,level))
                if curr.left:
                    queue.append((curr.left,level))


        return outputList

File: Javascript/BinaryTreeRightSideView.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0:val
    this.left = (left === undefined) ? null:left
    this.right = (right === undefined) ? null:right
}

var binaryTreeRightSideView = function(root){
    const outputArray = []
    const outputQueue = []
    const levelSet = new Set()
    

    let level = 0

    if(root !== null){
        outputQueue.push([root,level])
    }

    while(outputQueue.length>0){
        level += 1
        const lengthQueue = outputQueue.length
        for(let i=0;i<lengthQueue;i++){
            const [curr,levelToCheck] = outputQueue.shift()
            if(!levelSet.has(levelToCheck)){
                outputArray.push(curr.val)
                levelSet.add(levelToCheck)
            }

            if(curr.right){
                outputQueue.push([curr.right,level])
            }
            if(curr.left){
                outputQueue.push([curr.left,level])
            }
        }
    }

    return outputArray
}